targettests/Remote-Sim/observe_async.cpp

```diff
@@ -22,7 +22,16 @@ struct ansatz {
     cudaq::qvector q(2);
     x(q[0]);
     ry(theta, q[1]);
+// Note:
+// (1) The x<cudaq::ctrl> syntax requires C++-20.
```

Comment:
Since we compile the code with our own clang build, and this construct is eliminated during compilation, this syntax should be perfectly usable also when C++17 is requested. In face, nvq++ suppresses the warning about this being a C++20 feature by default. Did you run into any issues using it? 
In any case, the test should use x<cudaq::ctrl> regardless of C++ standard.

Reply:
> nvq++ suppresses the warning about this being a C++20 feature by default. 

In fact, this is the thing we want to check. Some of our public headers (e.g., `observe.h` or `sample.h`) have `#ifdef` switches for C++17 vs. C++20 compilation. Due to the default `-Wno-c++20-extensions` flag of `nvq++`, we didn't verify that the C++17 paths in these headers are actually free from C++20 code.  

In these test files, I added a specific `RUN` with `nvq++ -std=c++17 -Wc++20-extensions -Werror ` to verify that these C++17 paths in our public headers only contain C++17 code.